mllib/src/test/scala/org/apache/spark/mllib/pmml/export/DecisionTreePMMLModelExportSuite.scala

```diff
@@ -0,0 +1,364 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.pmml.export
+
+import scala.collection.JavaConverters._
+
+import org.dmg.pmml._
+import org.dmg.pmml.CompoundPredicate.BooleanOperator
+import org.scalatest.PrivateMethodTester
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.mllib.tree.configuration.{Algo, FeatureType}
+import org.apache.spark.mllib.tree.model.{DecisionTreeModel, Node, Predict, Split}
+import org.apache.spark.mllib.util.MLlibTestSparkContext
+
+class DecisionTreePMMLModelExportSuite extends SparkFunSuite
+with MLlibTestSparkContext
+with PrivateMethodTester {
+
+  test("PMML export should work as expected for DecisionTree model with regressor") {
+
+    // instantiate a MLLib DecisionTreeModel with Regression and with 3 nodes with continuous
+    // feature type
+    val mlLeftNode = new Node(2, new Predict(0.5, 0.5), 0.2, true, None, None, None, None)
+    val mlRightNode = new Node(3, new Predict(1.0, 0.5), 0.2, true, None, None, None, None)
+    val split = new Split(100, 10.00, FeatureType.Continuous, Nil)
+    val mlTopNode = new Node(1, new Predict(0.0, 0.1), 0.2, false,
+      Some(split), Some(mlLeftNode), Some(mlRightNode), None)
+
+    val decisionTreeModel = new DecisionTreeModel(mlTopNode, Algo.Regression)
+
+    // get the pmml exporter for the DT and verify its the right exporter
+    val pmmlExporterForDT = PMMLModelExportFactory.createPMMLModelExport(decisionTreeModel)
+    assert(pmmlExporterForDT.isInstanceOf[DecisionTreePMMLModelExport])
```

Comment:
This test on the right exporter could be moved to PMMLModelExportFactorySuite where I kept all the exporter checks for the factory.